homeassistant/components/switchbot/entity.py

```diff
@@ -4,43 +4,70 @@
 from collections.abc import Mapping
 from typing import Any
 
+from homeassistant.components import bluetooth
+from homeassistant.const import ATTR_CONNECTIONS
+from homeassistant.core import callback
 from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.entity import DeviceInfo, Entity
-from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import MANUFACTURER
-from .coordinator import SwitchbotDataUpdateCoordinator
+from .coordinator import SwitchbotCoordinator
 
 
-class SwitchbotEntity(CoordinatorEntity[SwitchbotDataUpdateCoordinator], Entity):
+class SwitchbotEntity(Entity):
     """Generic entity encapsulating common features of Switchbot device."""
 
+    coordinator: SwitchbotCoordinator
+
     def __init__(
         self,
-        coordinator: SwitchbotDataUpdateCoordinator,
-        idx: str | None,
-        mac: str,
+        coordinator: SwitchbotCoordinator,
+        unique_id: str,
+        address: str,
         name: str,
     ) -> None:
         """Initialize the entity."""
-        super().__init__(coordinator)
+        self.coordinator = coordinator
         self._last_run_success: bool | None = None
-        self._idx = idx
-        self._mac = mac
+        self._unique_id = unique_id
+        self._address = address
         self._attr_name = name
         self._attr_device_info = DeviceInfo(
-            connections={(dr.CONNECTION_NETWORK_MAC, self._mac)},
+            identifiers={(bluetooth.DOMAIN, self._address)},
```

Comment:
Normally we don't use another domain in identifiers.

Reply:
Maybe we should add a CONNECTION_BLUETOOTH to the device registry so integrations that use the same device could link them?

Reply:
https://github.com/home-assistant/core/pull/75666

I'll go back and refactor the coordinator in a followup as well to use it

Reply:
Yeah, adding a Bluetooth connection sounds good.

Reply:
Adjusted